spec/models/split_spec.rb

```diff
@@ -282,45 +282,173 @@
+    it "doesn't include a split with a remote kill spanning version" do
+      split = FactoryBot.create(:split)
+      FactoryBot.create(:app_remote_kill, split: split, app: app, first_bad_version: "0.8", fixed_version: "2.0")
+
+      expect(Split.excluding_remote_kills_for(app_build)).not_to include(split)
     end
 
-    it "returns the column value if it isn't feature-incomplete" do
-      fc = FactoryBot.create(:app_feature_completion, version: "1.1", split: subject)
-      app_build = fc.app.define_build(built_at: Time.zone.now, version: "1.1")
+    it "is backed by AppRemoteKill.affecting" do
+      allow(AppRemoteKill).to receive(:affecting).and_call_original
 
-      subject_with_knockouts = Split.with_feature_incomplete_knockouts_for(app_build).find(subject.id)
+      Split.excluding_remote_kills_for(app_build)
 
-      expect(subject_with_knockouts).not_to be_feature_incomplete
-      expect(subject_with_knockouts.registry).to eq("true" => 50, "false" => 50)
+      expect(AppRemoteKill).to have_received(:affecting).with(app_build)
     end
+  end
 
-    context "with a feature gate missing a false variant (gasp!)" do
-      subject { FactoryBot.create(:split, registry: { true: 50, not_false: 50 }, feature_gate: true) }
+  describe "#registry" do
+    context "with a feature gate" do
+      subject { FactoryBot.create(:split, registry: { true: 50, false: 50 }, feature_gate: true) }
+
+      it "returns the column value if it doesn't respond to feature_incomplete?" do
+        expect(subject).not_to respond_to(:feature_incomplete?)
+        expect(subject.registry).to eq("true" => 50, "false" => 50)
+      end
 
-      it "returns the column value and logs an error if feature-incomplete" do
+      it "overrides to 100% false if feature-incomplete" do
         fc = FactoryBot.create(:app_feature_completion, version: "1.1", split: subject)
         app_build = fc.app.define_build(built_at: Time.zone.now, version: "1.0")
 
-        subject_with_knockouts = Split.with_feature_incomplete_knockouts_for(app_build).find(subject.id)
-        allow(subject_with_knockouts.logger).to receive(:error).and_call_original
+        subject_with_knockouts = Split.for_app_build(app_build).find(subject.id)
+        expect(subject_with_knockouts).to be_feature_incomplete
+        expect(subject_with_knockouts.registry).to eq("true" => 0, "false" => 100)
+      end
 
+      it "returns the column value if it isn't feature-incomplete" do
+        fc = FactoryBot.create(:app_feature_completion, version: "1.1", split: subject)
+        app_build = fc.app.define_build(built_at: Time.zone.now, version: "1.1")
+
+        subject_with_knockouts = Split.for_app_build(app_build).find(subject.id)
+
+        expect(subject_with_knockouts).not_to be_feature_incomplete
+        expect(subject_with_knockouts.registry).to eq("true" => 50, "false" => 50)
+      end
+
+      it "prefers a remote kill over a feature incompletion if both are present" do
```

Comment:
Note that in the design doc, I was thinking that remote kills would conflict with feature incompletions, but I'm now realizing that you might want to override to a specific variant over a not-yet-released feature. For example, a feature flag with nonstandard variant names you realize too late. So now we allow them not to conflict and give remote kills precedence over feature completions. remote kills can't overlap one another, though, because it'd be impossible to tell which variant to override to without some annoying unsemantic tiebreaker logic.